### PR TITLE
Smugmug move

### PR DESCRIPTION
**Pull request by olsona**
Starting SmugMug move to new structure.  Not sure how we're going to do auth, so I abstracted out a SmugMugInterface (imitating the Google People or Calendar interfaces).

### CODE REVIEW

**Comment by rtannenbaum** [comment]
I am not very familiar with the auth extensions, could you ask someone more familiar with that to review? I didn't understand which Google people or calendar interfaces SmugMugInterface was based off, I don't think I have enough context.

File: extensions/data-transfer/portability-data-transfer-smugmug/build.gradle [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+dependencies {
+    compile project(':portability-spi-cloud')
+    compile project(':portability-spi-transfer')
+
+    compile("com.google.api-client:google-api-client:${googleApiClient}")
```
[the remark this answers is not in the document]
Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-smugmug/build.gradle]: Ahh this is for the HtppClient stuff - got it!

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugAlbumInfoResponse;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugResponse;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugUserResponse;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugAlbumsResponse;
+
+public class SmugMugInterface {
```
Comment (seehamrun) [inline extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java]: Class level comment

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(AuthData authData,
+      ExportInformation exportInformation) {
+    return null;
+  }
+
+  private ExportResult<PhotosContainerResource> exportAlbums(
+      Optional<PaginationData> paginationData) {
+    try {
```
Comment (seehamrun) [inline extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java]: can we wrap the try/catch around just the part that throws this exception (if its only a few lines long) - would make this easier to read imo

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(AuthData authData) {
+    return exportAlbums(Optional.empty());
+  }
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(AuthData authData,
+      ExportInformation exportInformation) {
+    return null;
```
Comment (seehamrun) [inline extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java]: Add TODO

File: settings.gradle [hunk 5]
```diff
@@ -28,3 +28,5 @@ include ':distributions:worker-default', ':distributions:gateway-default', ':dis
 
 include ':extensions:auth:portability-auth-flickr'
 include ':extensions:data-transfer:portability-data-transfer-flickr'
+
+include ':extensions:data-transfer:portability-data-transfer-smugmug'
```
Comment (seehamrun) [inline settings.gradle]: Can we hold off on including this in here until the TransferExtension and AuthServiceExtension are complete?

Comment (olsona) [reply settings.gradle]: Without that line, IntelliJ doesn't recognize portability-data-transfer-smugmug as a module.  Does that matter?

Comment (olsona) [reply settings.gradle]: Yeah, looks like I need to include the module for development reasons.

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java (hunk 2 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java]: Filled this in

File: settings.gradle (hunk 5 above)
Comment (seehamrun) [reply settings.gradle]: Oh youre right, sorry I thought this was the settings.gradle for the distributions.

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java (hunk 3 above)
Comment (olsona) [reply extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java]: Done.

**Pull request merged**